frappe/core/doctype/doctype/doctype.py

```diff
@@ -669,7 +669,8 @@ def validate_name(self, name=None):
 		flags = {"flags": re.ASCII} if six.PY3 else {}
 
 		# a DocType name should not start or end with an empty space
-		if re.match("^[ \t\n\r]+|[ \t\n\r]+$", name, **flags):
+		lead_trail_whitespace_re = re.compile("^[ \t\n\r]+|[ \t\n\r]+$", **flags)
+		if lead_trail_whitespace_re.findall(name):
```

Comment:
```suggestion
		if re.search("^[ \t\n\r]+|[ \t\n\r]+$", name, **flags):
```
even though `findall()` works, `search()` seems to be the right usage for this case. Also, compile isn't necessary if the pattern is not reused.

> If you want to locate a match anywhere in the string, use search() instead.
https://docs.python.org/3/library/re.html#re.match